draw_flip_book/lib/main.dart

```diff
@@ -50,25 +49,49 @@ class FlipBookPage extends StatefulWidget {
   _FlipBookPageState createState() => new _FlipBookPageState();
 }
 
-class _FlipBookPageState extends State<FlipBookPage> {
-  int currentFrame = 3;
+class _FlipBookPageState extends State<FlipBookPage>
+    with TickerProviderStateMixin {
+  AnimationController controller;
+
+  int currentFrame = 1;
   StrokeCap strokeCap = StrokeCap.round;
 
   // TODO: Generalize/Scale
-  bool _isVisible2 = true;
-  bool _isVisible3 = true;
+  bool _isVisible1 = true;
+  bool _isVisible2 = false;
+  bool _isVisible3 = false;
+  bool _isVisible4 = false;
+
+  bool _isAnimating = false;
+
+  double maxFrameOpacity = 0.7;
+
   bool _replayFrames = false;
 
   // TODO: Generalize/Scale into lists of List<Offset>
   List<Offset> points1 = List();
   List<Offset> points2 = List();
   List<Offset> points3 = List();
+  List<Offset> points4 = List();
 
   // TODO: Generalize/Scale
   // For accessing the RenderBox of each frame
-  GlobalKey key3 = GlobalKey();
-  GlobalKey key2 = GlobalKey();
   GlobalKey key1 = GlobalKey();
+  GlobalKey key2 = GlobalKey();
```

Comment:
I think you can name these keys more semantic names, e.g. `Key('frame2Key')`

Reply:
cleaned up the names of keys/cards